src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -8,10 +8,35 @@
 namespace System.Diagnostics
 {
     /// <summary>
-    /// Provides default implementation for Write and ShowDialog methods in Debug class.
+    /// Provides default implementation for Write and Fail methods in Debug class.
     /// </summary>
     public partial class DebugProvider
     {
+        public virtual void Fail(string message, string detailMessage)
+        {
+            string stackTrace;
+            try
+            {
+                stackTrace = new StackTrace(0, true).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
```

Comment:
Do the extra stackframes between the user code and this place show up in the stacktrace now?

Reply:
no they don't

Reply:
Right.